src/test/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModuleResolutionFunctionTest.java

```diff
@@ -365,4 +366,106 @@ public void testPrintBehavior() throws Exception {
     assertDoesNotContainEvent("hello from b@1.0");
     assertDoesNotContainEvent("hello from b@1.1");
   }
+
+  @Test
+  public void nodep_unfulfilled() throws Exception {
+    scratch.overwriteFile(
+        "MODULE.bazel",
+        """
+        bazel_dep(name='b',version='1.0')
+        bazel_dep(name='c',version='1.0',repo_name=None)
```

Comment:
How does it interact with dev_dependency = True, should we add some test cases?

Reply:
added a couple of cases. It's not super clear to me what would be a meaningful test case, but anyways...